### PR TITLE
Package the build_bundle.js script alongside ccfapp

### PR DESCRIPTION
**Pull request by cjen1-msft**
This PR exports build_bundle as an executable via `npx`, such that ACL applications can be more simply bundled.

The usage is `npx build_bundle path/to/acl/application` and writes a `bundle.json` to that directory.

Its also now converted over to typescript, simplified and the output of the governance proposal for `set_js_app` removed.

Comments on naming conventions, UX and testing are appreciated!

### CODE REVIEW

**Comment by cjen1-msft** [comment]
@microsoft-github-policy-service agree company="Microsoft"

**Comment by achamayou** [comment]
@cjen1-msft a CHANGELOG entry, and an update in the bit of doc about building JS app would be good to ensure this is discoverable by users. The sample will help too, of course.

File: js/ccf-app/src/scripts/build_bundle.ts [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    } else {
+      agg.push(filePath);
+    }
+  }
+  return agg;
+}
+
+function removePrefix(s: string, prefix: string): string {
+  if (s.startsWith(prefix)) {
+    return s.slice(prefix.length).split(sep).join(posix.sep);
```
Comment (achamayou) [inline js/ccf-app/src/scripts/build_bundle.ts:30]: Can this be made platform-independent by grabbing a more generic path separator?

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,6 +12,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ### Added
 
 - Expose `ccf:http::parse_accept_header()` and `ccf::http::AcceptHeaderField` (#6706).
+- Package `build_bundle.ts` under `npm exec build_bundle` to allow javascript users to build a ccf schema bundle (#6704).
```
Comment (achamayou) [inline CHANGELOG.md:15]: npx?

**Review by achamayou — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (cjen1-msft) [reply CHANGELOG.md:15]: Yeah this was a debate. I'm not sure which is more common in the npm space.

File: js/ccf-app/src/scripts/build_bundle.ts (hunk 1 above)
Comment (cjen1-msft) [reply js/ccf-app/src/scripts/build_bundle.ts]: Now resolved.
For posterity `path.sep` chooses the relevant separator depending on the operating system.

**Review by eddyashton — APPROVED** [review]

**Pull request merged**